### PR TITLE
WIP: Feature/improvements

### PR DESCRIPTION
**Pull request by cchet**
@Ollitod @marioaslan 
Please review the monthly report refactorings.

### CODE REVIEW

File: mega-zep-backend/src/main/java/com/gepardec/mega/service/impl/monthlyreport/calculation/journey/JourneyWarningCalculationStrategy.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import com.gepardec.mega.domain.model.monthlyreport.JourneyWarning;
+import com.gepardec.mega.domain.model.monthlyreport.ProjectTimeEntry;
+
+import java.time.LocalDate;
+import java.util.List;
+
+public interface JourneyWarningCalculationStrategy {
+
+    List<JourneyWarning> calculate(final LocalDate fromDate, final List<ProjectTimeEntry> projectTimeEntries);
```
Comment (Ollitod) [inline mega-zep-backend/src/main/java/com/gepardec/mega/service/impl/monthlyreport/calculation/journey/JourneyWarningCalculationStrategy.java:11]: Why is fromDate passed to calculation method?

Comment (Ollitod) [inline mega-zep-backend/src/main/java/com/gepardec/mega/service/impl/monthlyreport/calculation/journey/JourneyWarningCalculationStrategy.java:11]: Why is parameter fromDate passed to calculation method?

**Review by Ollitod — APPROVED** [review]
@cchet 

- unnecessary parameter fromDate in JourneyWarningCalculationStrategy (resolved)
- non-uniform names in concrete calculation strategies (resolved)

**Comment by cchet** [comment]
@Ollitod 
Copy Paste error. Only provide the list of projectTimes.
If a calculator needs grouping the calculator does itself. Take a look at the AbstractTimeWarningCalculationStrategy class, which encapsulates all common logic for the TimeWarning calculators.

**Comment by Ollitod** [comment]
@cchet 
Is resolved. I added a commit where I refactored the journey warning calculation. I am not quite sure about how to handle it best with the ResourceBundle and if calculation should be probably divided up further.

**Comment by cchet** [comment]
@Ollitod 
When the ResourceBundle is the only instance state in the WarningCalculator, then we can make the WarningCalculator an  @ApplicationScoped bean and inject the Resourcebundle.
This didn't work before because the old implementation had a state which was used by the public and private methods. Now we have no state anymore and the injected ResourceBundle is a CDI bean as well and is managed by a proxy which ensures that the invoked WarningCalculator public methods always get the ResourceBundle instance from the proper scope they are invoked in.

**Pull request merged**